src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
That said, I understand that we need to preserve the ordering of levels (not of `index`). So do anything that works, as long as you use the accessors.